### DISCUSSION TITLE
How to push mail credentials?

### DISCUSSION THREAD

**Discussion by sithumonline** [sliit-foss/rsvp#29] [Selection]
If we ignore it, it'll case to test the mail endpoint and in another hand it'll be security issue. so what we do???

**Comment by chamodshehanka** [comment]
I can remember azure has solution for this. can't remember that service name. so that GCP should have a solution as well.

**Reply by sithumonline** [reply]
ya GCP should

**Reply by chamodshehanka** [reply]
There's an easy way, we can pass the credentials as node ENVs with start script like this 
```
env NODE_ENV=local nodemon --exec 'ts-node' src/main.ts
```

So in your code first check the node envs are there else get them from config file.

**Reply by sithumonline** [reply]
```javascript 
export const MAIL = {
    HOST: process.env.MAIL_HOST || "smtp.mailtrap.io",
    PORT: process.env.MAIL_PORT || "2525",
    USER: process.env.MAIL_USER || "c8cca3941aea9c",
    PASSWORD: process.env.MAIL_PASSWORD || "18763bafd470a9",
}
```
Add something like above to `config/index.js` then import it to `mail.constants.js`, it'll solve both issues.

**Reply by sithumonline** [reply]
@susithrupasinghe can you implement this?

**Reply by susithrupasinghe** [reply]
> @susithrupasinghe can you implement this?

Yeah i can. I will

**Reply by susithrupasinghe** [reply]
@ebonynon  Fixed it now. #30

**Comment by susithrupasinghe** [comment]
Just give me a 30 minits. Still i'm on my bed.
